src/directives/domPlugin.ts

```diff
@@ -29,50 +19,103 @@ export class MDOMPlugin {
+        const plugin: IMElementPlugin<OptionsType> | undefined = MDOMPlugin.get(constructorFunction, element);
         if (plugin) {
             plugin.detach();
             delete element[constructorFunction.defaultMountPoint];
         }
     }
+
+    private static internalAttach<PluginType extends IMElementPlugin<OptionsType>, OptionsType>(constructorFunction: {
+        defaultMountPoint: string;
+        new (element: HTMLElement, options: OptionsType): PluginType;
+    }, element: HTMLElement, options: OptionsType): IMElementPlugin<OptionsType> {
+        let plugin: IMElementPlugin<OptionsType> = element[constructorFunction.defaultMountPoint] as IMElementPlugin<OptionsType>;
+        if (plugin) { MDOMPlugin.detach(constructorFunction, element); }
+
+        plugin = new constructorFunction(element, options);
+        if (plugin.status === MElementPluginStatus.Mounted) {
+            element[constructorFunction.defaultMountPoint] = plugin;
+            return plugin;
+        } else {
+            return new NullObjectMElementPlugin(options);
+        }
+    }
+
+    private static internalUpdate<PluginType extends IMElementPlugin<OptionsType>, OptionsType>(constructorFunction: {
+        defaultMountPoint: string;
+        new (element: HTMLElement, options: OptionsType): PluginType;
+    }, element: HTMLElement, options: OptionsType): PluginType {
+        const plugin: PluginType = element[constructorFunction.defaultMountPoint] as PluginType;
+        if (plugin) { plugin.update(options); }
+        return plugin;
+    }
 }
-export abstract class MElementPlugin<OptionsType> {
+
+export type MountFunction = (callback: () => void) => void;
+export interface IMElementPlugin<OptionsType> {
+    element: HTMLElement;
+    options: OptionsType;
+    status: MElementPluginStatus;
+    attach(mount: MountFunction): void;
+    update(options: any): void;
+    detach(): void;
+    addEventListener(eventName: string, listener: EventListenerOrEventListenerObject): void;
+    removeEventListener(eventName: string, listener?: EventListenerOrEventListenerObject): void;
+    removeAllEvents(): void;
+}
+
+class NullObjectMElementPlugin<OptionsType> implements IMElementPlugin<OptionsType> {
+    element: HTMLElement;
+    options: OptionsType;
+    status: MElementPluginStatus;
+    constructor(options: OptionsType) {
+        this.options = options;
+    }
+    attach(mount: MountFunction): void {}
+    update(options: any): void {}
+    detach(): void {}
+    addEventListener(eventName: string, listener: EventListenerOrEventListenerObject): void {}
+    removeEventListener(eventName: string, listener?: EventListener | EventListenerObject | undefined): void {}
+    removeAllEvents(): void {}
+}
+
+export enum MElementPluginStatus {
+    Mounted = 'mounted',
+    UnMounted = 'unmounted'
+}
+export abstract class MElementPlugin<OptionsType> implements IMElementPlugin<OptionsType> {
```

Comment:
J'ai jamais expliqué pk j'ai eu à faire ça mais voilà =>
Avec le drag and drop je dois être capable de gérer les 2 situations suivantes =>
1- pouvoir drag / drop / sort en associant une directive à l'élément
<div v-m-draggable/sortable/droppable></div>

2- je dois aussi être en mesure de "composer" dynamiquement le comportement d'une directive.  Pour éviter la duplication de code, fallait donc que j'extract la logique des directives sortable / droppable afin de pouvoir les composer dans sortable :)